src/components/AppointmentForm.tsx

```diff
@@ -46,6 +46,16 @@ const AppointmentForm = () => {
           setIsAppointmentDetailsPopupOpen(true);
           reset(defaultValues);
           setErrorMessage('');
+          emailHandler(createdAppointment)
```

Comment:
Looks good.

1. The appointment request details are mailed to the patient. 
2. The email send request is being handled as a server action. That's pretty neat.
3. The type of the form data sent is implicitly defined correctly.
4. The success and failure of the email send request promise has been handled.